backend/tests/Feature/Web/Admin/ProductRevisionTest.php

```
@@ -0,0 +1,242 @@
+
+        $this->actingAs($this->admin)->post(route('admin.work-orders.store'), [
+            'order_no' => 'WO-REV-2',
+            'product_type_id' => $type->id,
+            'product_revision_id' => $draft->id,
+            'planned_qty' => 10,
+        ])->assertSessionHasErrors('product_revision_id');
+    }
+
+    public function test_revision_change_is_blocked_after_production_starts(): void
+    {
+        $type = ProductType::factory()->create();
```

PHPStan (2.2.5)

[error] Call to an undefined static method App\Models\ProductType::factory().


(staticMethod.notFound)